lib/elixir/test/elixir/module/types/descr_test.exs

```diff
@@ -26,6 +26,12 @@ defmodule Module.Types.DescrTest do
       assert union(atom([:a]), negation(atom([:b]))) == negation(atom([:b]))
       assert union(negation(atom([:a, :b])), negation(atom([:b, :c]))) == negation(atom([:b]))
     end
+
+    test "dynamic" do
+      assert equal?(union(dynamic(), dynamic()), dynamic())
+      assert equal?(union(dynamic(), term()), term())
```

Comment:
Let's also add similar assertions but with the arguments flipped, for union, intersection, and difference.